classes/CartRule.php

```diff
@@ -321,6 +321,11 @@ public static function haveCartRuleToday($idCustomer)
             $sql .= 'UNION ALL (SELECT 1 FROM `' . _DB_PREFIX_ . 'cart_rule` ' .
                 'WHERE date_from >= "' . date('Y-m-d 00:00:00') .
                 '" AND date_from <= "' . date('Y-m-d 23:59:59') .
+                '" AND `id_customer` IN (0,' . (int)$idCustomer . ') LIMIT 1)';
```

Comment:
You can use a simple SQL request here : https://github.com/PrestaShop/PrestaShop/blob/develop/classes/CartRule.php#L319

Reply:
the new query is not cacheable by the query cache, and since there's no index on date_from / date_to, it scans a lot of rows from the ps_cart_rules table each time it executes the query (with the original query, it could be cached by the query cache a whole day).
From my point of view it's a performance regression if there's a lot of row in the ps_cart_rule table (and it happens for a lot of customers)